ocaml/xapi/xapi_cmd_result.mli

```diff
@@ -0,0 +1,30 @@
+(*
+ * Copyright (C) Citrix Systems Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+(* Module to parse result from command line output
+ * The result should be key value pairs seperated by sep like ':' *)
+
+type t
+
+(* Empty result set *)
+val empty : t
+
+(* Add one line to the result set *)
+val touch : ?sep:char -> string -> t -> t
+
+(* Find result from the result set *)
+val find : string -> t -> string option
```

Comment:
Are you using any of these outside of the context of the the `xapi_cmd_result.ml` file? If not, you can remove them from the .mli file. The idea of the .mli file is that you only expose what is necessary, in order to provide a clean interface

Reply:
some function is not used by me, like `touch` and `find`,  
but I take it as general module and provide functions that is necessary for future usage.

Reply:
The `find` is what currently I used, and it extract a value from the all output for `key`
However, there are a chance, that the output contains lots of key value pairs,  and we may want to query several values basing on `key`s,  and we do not want to execute the command several times.

In this case, `add` and `empty` will be useful to let the caller parses line by line.